docs/api/doc-block-useof.md

```diff
@@ -0,0 +1,109 @@
+
+Here’s an example of how the`useOf` hook could be used to create a custom block that displays the name of the story:
+
+```jsx
+// .storybook/blocks/StoryName.jsx
+
+import { useOf } from '@storybook/blocks';
+
+/**
+ * A block that displays the story name or title from the of prop
+ * - if a story reference is passed, it renders the story name
+ * - if a meta reference is passed, it renders the stories' title
+ * - if nothing is passed, it defaults to the primary story
+ */
+export const StoryName = ({ of }) => {
+  const resolvedOf = useOf(of || 'story', ['story', 'meta']);
+  switch (resolvedOf.type) {
+    case 'story': {
+      return <h1>{resolvedOf.story.name}</h1>;
+    }
+    case 'meta': {
+      return <h1>{resolvedOf.preparedMeta.title}</h1>;
+    }
+  }
+  return null;
+};
+```
+
+<!-- prettier-ignore-start -->
+```md
+{/* ButtonDocs.mdx */}
+import { Meta } from '@storybook/blocks';
+import { StoryName } from '../.storybook/blocks/StoryName';
+import * as ButtonStories from './Button.stories';
+
+<Meta of={ButtonStories} />
+
+{/* renders "Secondary" */}
+<StoryName of={ButtonStories.Secondary} />
+
+{/* renders "Primary" */}
+<StoryName />
+
+{/* renders "Button" */}
+<StoryName of={ButtonStories} />
+```
+<!-- prettier-ignore-end -->
+
+## useOf
+
+## Signature
+
+```ts
+useOf = (
+  moduleExportOrType: ModuleExport | 'story' | 'meta' | 'component',
+  validTypes?: Array<'story' | 'meta' | 'component'>
+): EnhancedResolvedModuleExportType
+```
+
+## Parameters
+
+### `moduleExportOrType` (required)
+
+Type: `ModuleExport | 'story' | 'meta' | 'component'`
+
+Provides the story export, meta export, component export, or CSF file exports that you want to get annotations from.
+
+When the custom block is in an [attached doc](./doc-block-meta.md#attached-vs-unattached), it’s also possible to get the primary (first) story, meta, or component by passing in a string instead. This is useful as a fallback, so the `of` prop can be omitted in your block. The most common pattern is using this as `useOf(props.of || 'story')` which will fall back to the primary story if no `of` prop is defined.
+
+- `useOf('story')` returns the annotated primary story in attached mode, error in unattached mode
+- `useOf('meta')` returns the annotated meta in attached mode, error in unattached mode
+- `useOf('component')` returns the annotated component specified in the meta in attached mode, error in unattached mode
+
+### `validTypes`
+
+Type: `Array<'story' | 'meta' | 'component'>`
+
+Optionally specify an array of valid types that your block accepts. Ie. the Canvas block only accepts references to stories, not meta or component, so it has `useOf(of, ['story'])` . If anything else than the valid types are passed in, an error will be thrown.
```

Comment:
When able rephrase this to "Optionally specify an array of valid types that your block accepts. Anything else that the valid types will result in an error. For example, the `Canvas` doc only accepts references to stories, not meta or component definitions. With that, it uses a `useOf(of, ['story'])` hook implementation." for a bit more clarity.